sys/net/sock/sock_util.c

```diff
@@ -116,30 +117,35 @@ static char* _find_pathstart(const char *url)
 
 int sock_urlsplit(const char *url, char *hostport, char *urlpath)
 {
+    assert(url);
     char *hoststart = _find_hoststart(url);
     if (!hoststart) {
         return -EINVAL;
     }
 
     char *pathstart = _find_pathstart(hoststart);
 
-    size_t hostlen = pathstart - hoststart;
-    /* hostlen must be smaller SOCK_HOSTPORT_MAXLEN to have space for the null
-     * terminator */
-    if (hostlen > SOCK_HOSTPORT_MAXLEN - 1) {
-        return -EOVERFLOW;
+    if (hostport) {
+        size_t hostlen = pathstart - hoststart;
+        /* hostlen must be smaller SOCK_HOSTPORT_MAXLEN to have space for the null
+        * terminator */
+        if (hostlen > SOCK_HOSTPORT_MAXLEN - 1) {
+            return -EOVERFLOW;
+        }
+        memcpy(hostport, hoststart, hostlen);
+        *(hostport + hostlen) = '\0';
     }
-    memcpy(hostport, hoststart, hostlen);
-    *(hostport + hostlen) = '\0';
 
-    size_t pathlen = strlen(pathstart);
-    if (pathlen) {
-        if (pathlen > SOCK_URLPATH_MAXLEN - 1) {
-            return -EOVERFLOW;
+    if (urlpath) {
+        size_t pathlen = strlen(pathstart);
+        if (pathlen) {
```

Comment:
This outer "if" is not necessary.

Reply:
*I think* removing the `if (pathlen) {` statement introduces possible undefined behaviour with the `memcpy()` (in the case `pathlen` equals `0`) below, unless you can show that `pathlen` here is always greater than `0`.

Reply:
I can't see the issue here, isn't it valid to pass 0 length to memcpy?

Reply:
Never mind, I think I confused myself here. IIRC it is undefined to call it with `NULL` as one of the arguments, even when the passed length is 0, but that's not the case here.